Caltech-IPAC/firefly
# FIREFLY-1353: Add arrow buttons for image scrolling in ImageTableRowViewer

Fixes [FIREFLY-1353](https://jira.ipac.caltech.edu/browse/FIREFLY-1353)

- Added react-slick slider to use in ImageTableRowViewer. slick-carousel package is also added for the css needed for react-slick
- Expose layout/container creating function in MultiItemViewerView so that items (images) can be rendered inside slider
- Synchronized changes in slide with change in highlighted row and vice versa (note: this is where logic gets bit complicated because we have to prevent it from looping)
- Created a custom slider arrow button for above purpose and to cancel the event when slider is hitting the left edge or right edge (so that it doesn't slide and active image moves to center)
- Made change in active plot through UI, change the highlighted row 
- Make height of slider (and inner components) take height of its parent container - that's why CSS sheet was added to override default slick styles

## Testing
https://firefly-1353-arrow-btn-imagerowviewer.irsakudev.ipac.caltech.edu/applications/spherex/

Make sure spectrum table is open - it's easier to test that way

- Click arrows, it should change the highlighted row. Note that slider won't really move until the active image comes to centre. Afterwards it will slide the slider along with moving the active image by one.
- Click on any row in table (or point in plot), it should change the slider position and active image. Also try changing the table pages.
-  Click on any image that is not active, it should change the highlighted row ~and move the slider to bring to center.~ preserving the slider position. On subsequent arrow clicks, test that it will move active plot/row or slider to bring it to center.
